### PR TITLE
SL-613: the skeleton for the new children component

### PR DESCRIPTION
**Pull request by cioan**
Before I go and implement the functionality for this new page, just wanted to make sure the component names and page name is acceptable to everyone:

Screenshot 2024-05-10 at 9 39 07 AM

Screenshot 2024-05-10 at 9 39 28 AM

### CODE REVIEW

File: omod/src/main/webapp/pages/patientRegistration/children.gsp [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+
+%>
+
+${ ui.includeFragment("coreapps", "patientHeader", [ patient: patient ]) }
+
+<script type="text/javascript">
+    var breadcrumbs = [
+        { icon: "icon-home", link: '/' + OPENMRS_CONTEXT_PATH + '/index.htm' },
+        { label: "${ ui.escapeJs(ui.format(patient.patient)) }" , link: '${ui.pageLink("pihcore", "router/programDashboard", ["patientId": patient.id])}'},
+        { label: "${ ui.message("pihcore.encounterList") }" , link: '${ui.pageLink("pihcore", "patient/encounterList", ["patientId": patient.id])}'}
```
Comment (mseaton) [inline omod/src/main/webapp/pages/patientRegistration/children.gsp:13]: Will need to clean up this reference - I assume this came from the page you copied to create this?

File: omod/src/main/webapp/pages/patientRegistration/children.gsp [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+    </thead>
+    <tbody>
+        <tr>
+            <td colspan="6">${ ui.message("emr.none") }</td>
+        </tr>
+    </tbody>
+</table>
+
+<div>
+    <input id="return-button" type="button" class="cancel" value="${ ui.message("pihcore.encounterList.return") }"/>
```
Comment (mseaton) [inline omod/src/main/webapp/pages/patientRegistration/children.gsp:79]: Same here, but I guess it's just a message code, not sure why it's specific to "Encounter List".

File: api/src/main/java/org/openmrs/module/pihcore/apploader/CustomAppLoaderFactory.java [hunk 3]
```diff
@@ -1306,6 +1310,15 @@ private void enableArchives() {
         addPaperRecordActionsIncludesIfNeeded();
     }
 
+    public void enableChildren() {
+        extensions.add(overallAction(CustomAppLoaderConstants.Extensions.PATIENT_CHILDREN_OVERALL_ACTION,
+                "registration.patient.children.label",
+                "fas fa-fw fa-child",
+                "link",
+                "pihcore/patientRegistration/children.page?patientId={{patient.patientId}}",
```
Comment (mseaton) [inline api/src/main/java/org/openmrs/module/pihcore/apploader/CustomAppLoaderFactory.java:1318]: Not 100% sure about nesting this under patientRegistration.  I can see the argument, but still not 100% sure.

Comment (mogoodrich) [reply api/src/main/java/org/openmrs/module/pihcore/apploader/CustomAppLoaderFactory.java:1318]: Yeah, I can see the argument as well, but let's just create a new "pihcore/children" folder instead.

File: omod/src/main/webapp/pages/patientRegistration/children.gsp [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+<script type="text/javascript">
+    var breadcrumbs = [
+        { icon: "icon-home", link: '/' + OPENMRS_CONTEXT_PATH + '/index.htm' },
+        { label: "${ ui.escapeJs(ui.format(patient.patient)) }" , link: '${ui.pageLink("pihcore", "router/programDashboard", ["patientId": patient.id])}'},
+        { label: "${ ui.message("pihcore.encounterList") }" , link: '${ui.pageLink("pihcore", "patient/encounterList", ["patientId": patient.id])}'}
+    ];
+
+    jq(document).ready(function() {
+        jq("#return-button").click(function(event) {
+            document.location.href = '${ui.pageLink("pihcore", "router/programDashboard", ["patientId": patient.id])}';
```
Comment (mogoodrich) [inline omod/src/main/webapp/pages/patientRegistration/children.gsp:18]: We should just return to the passed in "returnUrl" here...

**Pull request merged**